### PR TITLE
Authenticate Sequester using the org GitHub app token

### PR DESCRIPTION
**Pull request by BillWagner**
Update the code to use the org-configured GitHub app. 

Contributes to #177 

First, create a GitHub client that uses the org-level AppID and GitHub app secret. Write this code so that if the necessary secret isn't configured, fallback to the existing PAT. 

Update the query to read issues to additionally read the associated project and project fields. Those fields are only returned when using the updated app token. The code here optionally fills in those fields in the GitHub issue object if they are present. If not, the list of story point nodes is empty.

### CODE REVIEW

File: DotNet.DocsTools/GitHubClientServices/GitHubAppClient.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    internal async Task GenerateTokenAsync()
+    {
+        var privateKeySource = new PlainStringPrivateKeySource(_oauthPrivateKey);
+        var generator = new GitHubJwtFactory(
+            privateKeySource,
+            new GitHubJwtFactoryOptions
+            {
+                AppIntegrationId = _appID,
+                ExpirationSeconds = 8 * 60 // 600 is apparently too high
```
Comment (IEvangelist) [inline DotNet.DocsTools/GitHubClientServices/GitHubAppClient.cs:35]: So, we're able to request how long the token is alive before expiring? If so how does this hardcoded value correlate to the hardcoded five mins time span below?

File: DotNet.DocsTools/GitHubClientServices/GitHubClientBase.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+namespace DotNetDocs.Tools.GitHubCommunications;
+
+public abstract class GitHubClientBase : IGitHubClient, IDisposable
+{
+    private const string ProductID = "DotNetDocs.Tools";
+    private const string ProductVersion = "2.0";
+    private const string RESTendpoint = "https://api.github.com/repos";
+    private static readonly Uri graphQLUri = new("https://api.github.com/graphql");
+    private readonly HttpClient _client = new HttpClient();
```
Comment (IEvangelist) [inline DotNet.DocsTools/GitHubClientServices/GitHubClientBase.cs:16]:
```suggestion
    private readonly HttpClient _client = new();
```

File: DotNet.DocsTools/GitHubClientServices/GitHubClientBase.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            throw new InvalidOperationException(errorList.GetRawText());
+        else
+            return root.GetProperty("data");
+    }
+
+    async Task<JsonDocument> IGitHubClient.GetReposRESTRequestAsync(params string[] restPath)
+    {
+        var url = RESTendpoint;
+        foreach (var component in restPath)
+            url += "/" + component;
```
Comment (IEvangelist) [inline DotNet.DocsTools/GitHubClientServices/GitHubClientBase.cs:84]:
```suggestion
            url += %"/{component}";
```

File: actions/sequester/Quest2GitHub/Models/GithubIssue.cs [hunk 4]
```diff
@@ -205,6 +236,37 @@ public class GithubIssue
...
+
+                    var size = (sizeNode.ValueKind is JsonValueKind.String) ? sizeNode.GetString() : null;
+                    if ((projectTitle is not null) && (size is not null))
+                    {
+                        storyPoints.Add(new StoryPointSize(projectTitle, size));
+                    }
+                }
+            }
+        }
+        var closedEvent = issueNode.Descendent("timelineItems", "nodes").EnumerateArray()
```
Comment (IEvangelist) [inline actions/sequester/Quest2GitHub/Models/GithubIssue.cs:262]: Hate to think of edge cases like this, but what if it's closed and then reopened? Wouldn't this check need to be ordered by descending date, and then check if the last one is closed?

**Review by IEvangelist — APPROVED** [review]
Looks good, just have a few minor nits and questions.

File: DotNet.DocsTools/GitHubClientServices/GitHubClientBase.cs (hunk 3 above)
Comment (BillWagner) [reply DotNet.DocsTools/GitHubClientServices/GitHubClientBase.cs:84]: What is that `%`?  Did you mean `$`?

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**